### PR TITLE
refactor: moves user registraction logic under user controller

### PR DESCRIPTION
**Pull request by stalniy**
## Why

To make logic more transparent and add tests on db level to cover different edge cases. Ref: #1735

## Summary by CodeRabbit

* **New Features**
  * Added POST /v1/register-user and GET /v1/user/me (included in OpenAPI) and a public User schema.

* **Refactor**
  * Auth callback now registers/refreshes users and merges profile data.
  * Frontend onboarding uses a centralized storage key and injected URL service; logout now uses in-app logout that clears tokens and redirects.

* **Tests**
  * New comprehensive tests for user registration, wallet/anonymous flows, and auth interceptor behavior.

* **Documentation**
  * Deprecated old token info route in favor of new endpoints.

### CODE REVIEW

File: apps/api/src/user/services/user/user.service.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  let baseUsername = wantedUsername.replace(/[^\w-]+/g, "");
+
+  if (baseUsername.length < 3) {
+    baseUsername = baseUsername.padEnd(10, "anonymous");
+  } else if (baseUsername.length > 40) {
+    baseUsername = baseUsername.slice(0, 40);
+  }
+
+  return baseUsername + randomInt(1000, 9999).toString();
+}
```
[the remark this answers is not in the document]
Comment (stalniy) [reply apps/api/src/user/services/user/user.service.ts:126]: sensible defaults may cause username unique index conflicts

File: apps/api/src/core/repositories/base.repository.ts [hunk 2]
```diff
@@ -221,3 +222,7 @@ type TableName<T extends PgTableWithColumns<any>> = T extends PgTableWithColumns
 type TableNameInSchema<T extends PgTableWithColumns<any>> = {
   [K in keyof TablesOnly<ApiPgTables> as TableName<ApiPgTables[K]>]: K;
 }[TableName<T>];
+
+export function isUniqueViolation(error: unknown): error is PostgresError {
+  return error instanceof PostgresError && error.code === "23505";
```
Comment (baktun14) [inline apps/api/src/core/repositories/base.repository.ts]: suggestion(non-blocking): use a const for the code

File: apps/api/src/user/routes/register-user/register-user.router.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+export type RegisterUserInput = z.infer<typeof registerUserInputSchema>;
+
+const registerUserResponseSchema = z.object({
+  data: UserSchema
+});
+export type RegisterUserResponse = z.infer<typeof registerUserResponseSchema>;
+
+const route = createRoute({
+  method: "post",
+  path: "/v1/registerUser",
```
Comment (baktun14) [inline apps/api/src/user/routes/register-user/register-user.router.ts]: suggestion(non-blocking): we use lower case dash for new routes
```suggestion
  path: "/v1/register-user", // or /v1/register might be simpler
```

File: apps/api/src/routers/userRouter.ts [hunk 4]
```diff
@@ -39,6 +39,9 @@ userOptionalRouter.get("/byUsername/:username", async c => {
   return c.json(user);
 });
 
+/**
+ * @deprecated Use /v1/registerUser instead
```
Comment (baktun14) [inline apps/api/src/routers/userRouter.ts]: reminder(non-blocking): in case you change the route

**Review by baktun14 — DISMISSED** [review]

File: apps/api/src/core/repositories/base.repository.ts (hunk 2 above)
Comment (stalniy) [reply apps/api/src/core/repositories/base.repository.ts]: do we have a const for this? or do you want me to create one? I thought that this method is small and does exactly what it does, checks whether the passed in error is a unique violation error from postgres.

Comment (baktun14) [reply apps/api/src/core/repositories/base.repository.ts]: Create one, but I guess the name of the function tells what it does
